.github/workflows/ruff.yml

```diff
@@ -0,0 +1,16 @@
+name: ruff
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  ruff:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - run: pip install --user ruff
+    - run: ruff --ignore=E402,F401,F403,F405,F541,F601,F811,F841
+                --format=github --line-length=214 --target-version=py37 .
```

Comment:
As far as I understand, this replaces flake8, so it probably makes sense to remove the flake8 check if this will be used.

I wonder about that line length of 214 requirement. We also use `black`, which uses 88 as the maximum line length, which `ruff` also seems to use, so why this setting? In case it doesn't understand the `noqa` comment, it would be better to switch off the check for line length, as it is already done by `black` anyway.

Also, why all the exclusions? Apart from E402 (module-import-not-at-top-of-file) I see no reason for the rest.

Reply:
https://github.com/pydicom/pydicom/blob/master/Makefile#L32-L34

Reply:
Sorry, I don't understand what you mean. 
I can see that this makefile is quite outdated and should probably be adapted or removed, though :)

Reply:
Let's leverage `--format=github` to show where the errors are.

Reply:
Please review the first ten errors in the code listing... https://github.com/pydicom/pydicom/pull/1771/files

Let's agree that `line-length` is currently different under `psf/black` (best effort formatter) and `ruff` (rules-based linter) but ___our goal is to make them the same___.  The difference because the former is guidance for a tool that formats code __but not comments and is deliberately timid about reformatting strings__ while the latter is the trigger for one or more good-first-issues for contributors to shorten long lines that `psf/black` does not autofix.

Reply:
About the first ten errors: I think these are all things that can be either easily fixed, or these lines should get an exclusion comment. I just think it makes more sense to fix them instead of disabling the warnings.
As for line-length: as far as I could see, these are from generated code. It would make sense just to exclude these files from the check by a file-wide exclusion comment, if that is possible. The same is true for unused imports in `__init__.py`.

Anyway, I'm not saying that you have to do all of it now - that is something that can also be handled later by somebody else. In this case I would like to at least remove the redundant flake8 check (it is redundant, right?), and write a new issue for enabling and fixing these warnings.

Reply:
`pyproject.toml` added with appropriate settings and #1776 add to cover follow-on tasks.